proxy/coco/src/announcement.rs

```diff
@@ -96,15 +96,19 @@ mod test {
     use crate::error::Error;
     use crate::oid;
     use crate::peer;
+    use crate::signer;
 
     #[tokio::test]
     async fn announce() -> Result<(), Error> {
         let tmp_dir = tempfile::tempdir().expect("failed to create temdir");
         let key = SecretKey::new();
-        let config = config::default(key.clone(), tmp_dir.path())?;
+        let signer = signer::BoxedSigner::new(signer::SomeSigner {
+            signer: key.clone(),
+        });
+        let config = config::default(key, tmp_dir.path())?;
```

Comment:
Shouldn't the `config` functions take the `BoxedSigner` too? :thinking:

Reply:
Nvm

Reply:
For others, this is not part of this change as we still need to land some improvements in link. Until then we at least migrate the majority of the codebase to use signer.